### ISSUE TITLE
Is `obs_t` the right name? (for right truncation bound)

### ISSUE THREAD

**Issue by athowes** [epinowcast/epidist#321]
> Is `obs_t` the right name?

_Originally posted by @athowes in https://github.com/epinowcast/epidist/pull/315#discussion_r1756570387_

Similar to #320. This is the right truncation bound. I think that `obs_t` indicates it being an absolute time rather than relative to primary event time, so I don't like it in that sense at the moment.

Options:

* `D`
* `D_upr`

Want to do something that is extensible to lower bound. Maybe we could be using a vector for the domain of the function, could even be `domain = c(a, b)`.

Actually, $[a, b]$ is pretty normal notation for bounds, we could do that too, like `a` and `b`.

Other options here?

**Comment by seabbs** [comment]
I agree with the maths motivation but think we want to put our epi hats on here and try and name it based on that. @kgostic any thoughts?

**Comment by seabbs** [comment]
An incredibly boring choice could be `relative_obs_cutoff` or similar

**Comment by athowes** [comment]
Note that this package (despite being called `epidist`) might not only be suitable for epidemiology.

**Comment by athowes** [comment]
I think `a` and `b` are actually pretty good and of course we can document this for people.

**Comment by seabbs** [comment]
agree but that domain is the main target (and applied modellers more generally)

**Comment by athowes** [comment]
We can look for papers / textbooks / lecture notes on left and right truncation and use notation from there. Though of course that'll be short names not longer descriptive.

**Comment by seabbs** [comment]
We can but keep in mind above point about what is familiar to epis/applied users as I do think this is critical

**Comment by athowes** [comment]
Clarity on `obs_at` and `obs_t`:

* `obs_at` is absolute (relative to the first event)
* `obs_t` is relative

Could be `relative_obs_time` and `obs_time`. I'd be happy with this.

**Comment by seabbs** [comment]
Agree with the proposed names and that matches `primarycensoreddist`
